src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -98,22 +98,36 @@ public async Task SendRequestBodyAsync(CancellationToken cancellationToken)
                     {
                         using (Http2WriteStream writeStream = new Http2WriteStream(this))
                         {
-                            await _request.Content.CopyToAsync(writeStream, null, cancellationToken).ConfigureAwait(false);
+                            // TODO: until #9071 is fixed, cancellation on content.CopyToAsync does not work.
+                            // To work around it, register delegate and set _abortException as needed.
+                            using (cancellationToken.Register(stream => { if (((Http2Stream)stream)._abortException == null) ((Http2Stream)stream)._abortException = new OperationCanceledException(); }, this))
+                            {
+                                await _request.Content.CopyToAsync(writeStream, null, cancellationToken).ConfigureAwait(false);
+                            }
                         }
 
                         // Don't wait for completion, which could happen asynchronously.
-                        _ = _connection.SendEndStreamAsync(_streamId);
+                        _ = _connection.LogExceptionsAsync(_connection.SendEndStreamAsync(_streamId));
                     }
                     catch (Exception e)
                     {
-                         // We did not finish sending body so notify server.
-                         _ = _connection.SendRstStreamAsync(_streamId, Http2ProtocolErrorCode.Cancel);
+                         // Try to notify server if we did not finish sending request body.
+                         _ = IgnoreExceptionsAsync(_connection.SendRstStreamAsync(_streamId, Http2ProtocolErrorCode.Cancel));
 
                         // if we decided abandon sending request and we get ObjectDisposed as result of it, just eat exception.
-                        if (_shouldSendRequestBody || (!(e is ObjectDisposedException) && !(e.InnerException is ObjectDisposedException)))
+                        if (!_shouldSendRequestBody && (e is ObjectDisposedException || e.InnerException is ObjectDisposedException))
                         {
-                            throw;
+                            return;
                         }
+
+                        if (_abortException == null)
+                        {
+                            // If we still processing response after receiving response headers, this will give us chance to propagate exception up.
```

Comment:
Nit:
"If we still processing response" => "If we are still the response"
"this will give us chance" => "this will give us a chance"